Homeworks/Homework W8/SQLQuery_HomeworkW8.sql

```diff
@@ -0,0 +1,340 @@
+CREATE DATABASE ZooManagemnt;
+
+USE ZooManagemnt;
+-- Creating tables
+CREATE TABLE ZoneArea
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Zone VARCHAR(50)
+);
+
+CREATE TABLE Species
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Name VARCHAR(50),
+Zone INT REFERENCES ZoneArea(Id)
+);
+
+CREATE TABLE Animals
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Species INT REFERENCES Species(Id)
+);
+
+CREATE TABLE Workers
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+FirstName VARCHAR (20),
+LastName VARCHAR (20),
+WorkArea INT REFERENCES ZoneArea(Id)
+);
+
+CREATE TABLE Ticket
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Price MONEY,
+ValidityStart DATE,
+ValidityEnd DATE,
+Zone INT REFERENCES ZoneArea(Id)
```

Comment:
WorkAreaId instead of Zone. Consistency is really important in programming. Above, in the workers table, you have the exact same foreign key to a ZoneArea and it is called WorkArea.
So a column called WorkArea and a column called Zone are both referencing a ZoneArea - quite confusing when reading it, right? :)